File: regions/system/library/dart/websocket/lib/src/ws_message.dart
import 'dart:typed_data';

enum MessageType { text, binary, ping, pong, close }

class WsMessage {
  final MessageType type;
  final Object payload;

  const WsMessage({required this.type, required this.payload});

  String get textPayload => payload as String;
  Uint8List get binaryPayload => payload as Uint8List;
}

File: regions/system/library/csharp/websocket/src/K1s0.WebSocket/WsMessage.cs
namespace K1s0.WebSocket;

public enum MessageType
{
    Text,
    Binary,
    Ping,
    Pong,
    Close,
}

public record WsMessage(MessageType Type, ReadOnlyMemory<byte> Payload);

public enum ConnectionState
{
    Disconnected,
    Connecting,
    Connected,
    Reconnecting,
    Closing,
}
